core/sail/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneSail.java

```diff
@@ -576,78 +576,82 @@ public void setFederatedServiceResolver(FederatedServiceResolver resolver) {
 	 * Starts a reindexation process of the whole sail. Basically, this will delete and add all data again, a
 	 * long-lasting process.
 	 *
-	 * @throws IOException
+	 * @throws SailException If the Sail could not be reindex
 	 */
-	public void reindex() throws Exception {
-		// clear
-		logger.info("Reindexing sail: clearing...");
-		luceneIndex.clear();
-		logger.info("Reindexing sail: adding...");
-
-		luceneIndex.begin();
+	public void reindex() throws SailException {
```

Comment:
@jeenbroekstra I was wondering if we consider this a backwards compatible change or not? Japicmp doesn't seem to be complaining. If the change is ok with you, then I propose that we can remove the throws declaration all together since SailException is a RuntimeException.